test/file02.tsx

```diff
@@ -0,0 +1,114 @@
+import PoweredByGoogleIcon from '@xsite-ui/prequal-funnels.assets/powered-by-google';
+import { INPUT_TYPES } from '@xsite-ui/prequal-funnels.constants';
+import React, {
+  ChangeEventHandler,
+  FocusEventHandler,
+  useCallback,
+  // useEffect,
+  useState,
+} from 'react';
+
+import { TEXT_INPUT_ILLEGAL_CHARACTERS, TEXT_INPUT_MIN_LENGTH } from '../../../input/constants';
+import {
+  Container,
+  ErrorLabelStyle,
+  InfoLabelStyle,
+  PlaceHolder,
+  StyledInput,
+} from '../../../input/styles';
+import { PoweredByGoogle } from './styles';
+
+const FullAddress = ({ onBlur, question, stepAnswers }) => {
+  const { answers, config, id: questionId } = question;
+  const { profileKey = '' } = config ?? {};
+  const answer = answers?.[0] ?? {};
+  const { data, id: answerId } = answer;
+  const { errorText } = data ?? {};
+
+  const [isClicked, setIsClicked] = useState(false);
+  const [inputValue, setInputValue] = useState((stepAnswers?.[questionId]?.value as string) ?? '');
+  const [isInputFilled, setIsInputFilled] = useState(
+    Boolean(stepAnswers?.[questionId]?.value as string),
+  );
+  const [isInvalid, setIsInvalid] = useState(false);
+
+  const clickHandler = () => {
+    if (isClicked) {
+      return;
+    }
+
+    setIsClicked(true);
+  };
+
+  const inputHandler: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const { value = '' } = e?.target ?? {};
+
+    if (value && !TEXT_INPUT_ILLEGAL_CHARACTERS.test(value)) {
+      return;
+    }
+
+    setInputValue(value);
```

Comment:
Use if...else statement instead of return statement to avoid skipping setInputValue call when illegal characters are entered